projects/arlas-toolkit/src/lib/tools/collection-service-implementation.service.ts

```diff
@@ -0,0 +1,53 @@
+import { CollectionUnit } from './utils';
+import { map, Observable } from 'rxjs';
+import { BaseCollectionService } from 'arlas-web-components';
+import { Injectable } from '@angular/core';
+import { ArlasCollaborativesearchService, ArlasConfigService } from '../services/startup/startup.service';
+import { CollectionReferenceDescription } from 'arlas-api';
+import { CollectionDisplayNames } from 'arlas-api/api';
+
+@Injectable()
+export class CollectionServiceImplementation extends BaseCollectionService {
+  private appUnits: Map<string, CollectionUnit> = new Map();
+  public constructor(private collaborativeService: ArlasCollaborativesearchService, private configService: ArlasConfigService) {
+    super();
+    this._setUnits();
+  }
+
+  protected _setUnits() {
+    const appUnits = this.configService.getValue('arlas-wui.web.app.units') ?
+      this.configService.getValue('arlas-wui.web.app.units') : [];
+    appUnits.forEach((collectionUnit: CollectionUnit) => this.appUnits.set(collectionUnit.collection, collectionUnit));
+    /** retrocompatibility code for unit*/
+    const appUnit = this.configService.getValue('arlas-wui.web.app.unit');
+    if (appUnit || appUnits.length === 0) {
+      const collectionUnit = {
+        collection: this.collaborativeService.defaultCollection,
+        unit: !!appUnit ? appUnit : this.collaborativeService.defaultCollection,
+        ignored: false
+      };
+      this.appUnits.set(collectionUnit.collection, collectionUnit);
+    }
+    /** end of retrocompatibility code */
+  }
+
+  public getUnit(collectionName: string): string | null {
+    if (this.appUnits.has(collectionName)) {
+      return this.appUnits.get(collectionName).unit;
+    }
+    return null;
+  };
+
+  public getDisplayName(collectionName: string): Observable<CollectionDisplayNames | undefined> {
```

Comment:
In the setUnits, can't you fetch the display name of all the collections of the app ? This way you can store them in the same way as the units, and it will be easy to retrieve them

Reply:
Do you think its a good idea to loop collection names to store them ?

Reply:
Since they will be used in a pipe, having them already stored will make things smoother and ensure that there is only one call per collection

Reply:
as the requests are async, if sometime we do not have time to fetch all displayNames, we could run into errors.

Reply:
Yes, but if the requests are made during the intialization of the app, this problem is solved. In the worst case, the service could know that it is still loading some collections and based on that still return an observable that is either solved immediately if the key exists, or will just be resolved once the query has been answered. I do think that loading everything in the beginning is a better solution, similarly to the ArlasStartupService

Reply:
@MAudelGisaia @QuCMGisaia 
`ArlasStartupService` already fetches the descriptions of all collection at App initialisation and stores them in `collectionMap` attribute.
I suggest to ingest the `ArlasStartupService` in your service and use the `collectionMap` attribute. This way we avoid re-sending a request to arlas-server for an info we already have.
The only thing to check, though, before using this attribute, is whether `shouldRunApp` (in `ArlasStartupService`) is true